### PR TITLE
[Bugfix]Ignoring rejected events

### PR DESCRIPTION
**Pull request by mhib**
@adamzima @jtkpiotr @quirk0o @pbzymek @s1mplex

### CODE REVIEW

File: app/models/google_event.rb [hunk 1]
```diff
@@ -14,19 +14,21 @@ class GoogleEvent
...
         rooms.each do |room|
           config = {fields: FIELDS, single_events: true, time_min: starting.rfc3339(9),
-                    time_max: ending.rfc3339(9), time_zone: 'Europe/Warsaw'}
+                    time_max: ending.rfc3339(9), time_zone: 'Europe/Warsaw',
+                    always_include_email: true}
           service.list_events(room.email, config) do |result, _|
             next unless result
             result.items&.each do |event|
               normalize_dates(event)
+              next if event.attendees.find(&:self).response_status == 'declined'.freeze
```
Comment (pbzymek) [inline app/models/google_event.rb]: make `'declined'` a constat

File: app/models/google_event.rb [hunk 2]
```diff
@@ -14,19 +14,21 @@ class GoogleEvent
...
       rooms = ConferenceRoom.all
       calendar_service(credentials).batch do |service|
         rooms.each do |room|
           config = {fields: FIELDS, single_events: true, time_min: starting.rfc3339(9),
-                    time_max: ending.rfc3339(9), time_zone: 'Europe/Warsaw'}
+                    time_max: ending.rfc3339(9), time_zone: 'Europe/Warsaw',
+                    always_include_email: true}
           service.list_events(room.email, config) do |result, _|
             next unless result
             result.items&.each do |event|
```
Comment (pbzymek) [inline app/models/google_event.rb]: is '&' here on purpose ?

File: app/models/google_event.rb [hunk 3]
```diff
@@ -14,19 +14,21 @@ class GoogleEvent
 
   class << self
     # You can specify custom fields: https://developers.google.com/google-apps/calendar/v3/reference/events
-    FIELDS = 'items(id, start, end, summary, recurrence, creator)'.freeze
+    FIELDS = 'items(id, start, end, summary, recurrence, creator, attendees(self, responseStatus))'.freeze
```
Comment (pbzymek) [inline app/models/google_event.rb]: I'm not sure if building a constant base on a response each time is a good solution here

File: app/models/google_event.rb (hunk 1 above)
Comment (madejejej) [inline app/models/google_event.rb]: This looks like a duct tape with indentation level: 4. I think if you won't refactor this method now you will never be able to do so in the future :D

File: app/models/google_event.rb [hunk 4]
```diff
@@ -111,6 +105,20 @@ def load_emails
       ConferenceRoom.pluck(:email)
     end
 
+    GOOGLE_EVENT_DECLINED_RESPONSE = 'declined'.freeze
+    def add_events_from_room(room, service, events, config)
+      service.list_events(room.email, config) do |result, _|
+        if result
```
Comment (madejejej) [inline app/models/google_event.rb:124]: When can `result` be nil?

File: app/models/google_event.rb [hunk 5]
```diff
@@ -111,6 +105,20 @@ def load_emails
...
+          result.items.each do |event|
```
Comment (madejejej) [inline app/models/google_event.rb:125]: I noticed you changed `result.items&.each` to `result.items.each`. Can `result.items` be nil?

File: app/models/google_event.rb [hunk 6]
```diff
@@ -111,6 +105,20 @@ def load_emails
...
+            next if event.attendees.find(&:self).response_status == GOOGLE_EVENT_DECLINED_RESPONSE
```
Comment (madejejej) [inline app/models/google_event.rb]: I'd name the conditional. It also heavily violates Law of Demeter. Is `event` an object of class you implement or does it come from Google's implementation?

File: app/models/google_event.rb (hunk 5 above)
Comment (mhib) [inline app/models/google_event.rb:125]: I don't think so

File: app/models/google_event.rb (hunk 6 above)
Comment (mhib) [inline app/models/google_event.rb]: Event comes from Google.

File: app/models/google_event.rb [hunk 7]
```diff
@@ -111,6 +105,20 @@ def load_emails
...
+            normalize_dates(event)
+            events[event.start.date_time.wday] ||= []
+            events[event.start.date_time.wday] << event.to_h.merge(conference_room: room)
```
Comment (madejejej) [inline app/models/google_event.rb:128]: Any reason to use `Hash` instead of a custom object?

File: app/models/google_event.rb (hunk 4 above)
Comment (mhib) [inline app/models/google_event.rb:124]: When we do not have permissions from Google to load events from a given calendar.

File: app/models/google_event.rb (hunk 7 above)
Comment (mhib) [inline app/models/google_event.rb:128]: Serialization. Almost whole logic with returned object is done in JavaScript anyway.

File: app/models/google_event.rb (hunk 4 above)
Comment (madejejej) [inline app/models/google_event.rb:124]: Do you notify this to the user or is it a silent failure?

Comment (mhib) [inline app/models/google_event.rb:124]: Silent failure

**Pull request merged**